src/stores/useSocialMediaStore/useSocialMediaStore.spec.ts

```diff
@@ -0,0 +1,101 @@
+import { act, renderHook } from '@testing-library/react';
+
+import * as useSocialMediaStoreHook from '~stores/useSocialMediaStore/useSocialMediaStore';
+
+vi.mock('~services/api/social-media/social-media', () => ({
+  SocialMediaService: {
+    fetch: vi.fn().mockResolvedValue([
+      { icon: 'discord-icon', id: 'DISCORD_EXAMPLE_ID', name: 'Discord' },
+      { icon: 'twitter-icon', id: 'TWITTER_EXAMPLE_ID', name: 'Twitter' },
+    ]),
+  },
+}));
+
+vi.mock('~services/api/accounts/accounts', () => ({
+  AccountsService: {
+    fetchAll: vi.fn().mockResolvedValue([
+      {
+        avatar: 'https://example.com/image1.jpg',
+        expiresAt: 'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        generatedAt:
+          'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        id: 1,
+        socialMediaId: 'DISCORD_EXAMPLE_ID',
+        token: 'DISCORD_EXAMPLE_TOKEN_1',
+        userName: 'Discord User 1',
+        valid: true,
+      },
+      {
+        avatar: 'https://example.com/image2.jpg',
+        expiresAt: 'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        generatedAt:
+          'Fri May 17 2024 22:57:54 GMT-0300 (Brasilia Standard Time)',
+        id: 14,
+        socialMediaId: 'TWITTER_EXAMPLE_ID',
+        token: 'TWITTER_EXAMPLE_TOKEN_14',
+        userName: 'Twitter User 14',
+        valid: true,
+      },
+    ]),
+  },
+}));
+
+/*
+The getAllAccounts call is necessary because although the getAllAccounts function is 
+defined in the store body, it is not automatically executed when the state object is 
+created and the behavior we want is to be called automatically when the store is initialized.
+*/
```

Comment:
`in the store file`

`the getAllAccount dont mock wen called this way`

`so we need to manually call it to be able to mock on the tests below`

E ai coloca o link da sua conversa com o menino do zustand